project/SparkBuild.scala

```diff
@@ -1373,6 +1373,7 @@ object Unidoc {
     classpaths
       .map(_.filterNot(_.data.getCanonicalPath.matches(""".*kafka-clients-0\.10.*""")))
       .map(_.filterNot(_.data.getCanonicalPath.matches(""".*kafka_2\..*-0\.10.*""")))
+      .map(_.filterNot(_.data.getCanonicalPath.contains("apache-rat")))
```

Comment:
It seems that the `apache-rat-0.14.jar` downloaded during `check-license` is interfering with the generation of unidoc, because there is an old version of the `JavaVersion` class in `apache-rat-0.14.jar` that has not been relocated... Let me try if this change can make the unidoc build successful first. If it can, I will submit a separate PR to complete this ignore work first.

Reply:
Can we explicitly pin the version via `DependencyOverrides` if the version conflicts?

Reply:
The following contents have been shaded in apache-rat-0.14.jar, but none of them have been relocated:

```
org.apache.commons:commons-collections4:jar:4.4
org.apache.commons:commons-lang3:jar:3.12.0
commons-io:commons-io:jar:2.11.0
org.apache.commons:commons-compress:jar:1.25.0
commons-cli:commons-cli:jar:1.5.0
```

Since `apache-rat` is not a direct dependency of the project, it was manually downloaded in `check-license`, I did not find a way to exclude it through `DependencyOverrides` or `ExcludedDependencies`.

Another possible simple solution is to change `build_and_test.yml`, manually delete `./lib/apache-rat-*.jar` after executing `check-license`.

@HyukjinKwon WDYT?

Reply:
I have submitted a separate pr to address this issue: https://github.com/apache/spark/pull/44453. If the modifications to `SparkBuild.scala` can be done together in the current PR, please let me know.